### PR TITLE
Fix/react event callback types

### PR DESCRIPTION
**Pull request by cjpillsbury**
Also did some additional "under the hood" cleanup of types and vestigial code.

### CODE REVIEW

**Review by luwes — APPROVED** [review]
nice!

File: packages/mux-player-react/src/index.tsx [hunk 1]
```diff
@@ -68,42 +73,42 @@ export type MuxPlayerProps = {
...
+  onPlaying?: GenericEventListener<MuxPlayerElementEventMap['playing']>;
+  onTimeUpdate?: GenericEventListener<MuxPlayerElementEventMap['timeupdate']>;
+  onPause?: GenericEventListener<MuxPlayerElementEventMap['pause']>;
+  onSeeking?: GenericEventListener<MuxPlayerElementEventMap['seeking']>;
+  onSeeked?: GenericEventListener<MuxPlayerElementEventMap['seeked']>;
+  onStalled?: GenericEventListener<MuxPlayerElementEventMap['stalled']>;
+  onSuspend?: GenericEventListener<MuxPlayerElementEventMap['suspend']>;
+  onEnded?: GenericEventListener<MuxPlayerElementEventMap['ended']>;
+  onError?: GenericEventListener<MuxPlayerElementEventMap['error']>;
+  // onPlayerReady?: EventListener;
```
Comment (spenweb) [inline packages/mux-player-react/src/index.tsx:99]: @luwes why was it decided to remove the `onPlayerReady` prop and corresponding functionality?

Comment (luwes) [reply packages/mux-player-react/src/index.tsx:99]: @spenweb so far there was no need for this callback. The MuxPlayer is an inline JS embed, once the JS constructor executes the API is ready to accept calls. There are all the native video events available like `loadstart`, `loadedmetadata`, etc. which should provide enough indication when the src is set and loaded etc.

Do you have a need for an onPlayerReady callback?

Comment (spenweb) [reply packages/mux-player-react/src/index.tsx:99]: Yes, I need to be able to change the volume after the player UI has initialized. I've been experiencing odd behavior where I am able to change the volume but the volume UI slider does not update. This leads me to believe that I need to wait for the player UI to initialize before I can set the volume.

Here is my current work-around hack using a useEffect:
```ts
// Control the Mux player's volume state
useEffect(() => {
  // volume is coming from a prop
  targetVolumeRef.current = volume

  let timeoutId: NodeJS.Timeout | null = null
  let cancel: boolean

  const enforcer = () => {
    if (
      !muxPlayerRef.current ||
      cancel ||
      targetVolumeRef.current === undefined
    ) {
      return
    }
    // `updateVolumeEquilibrium` just checks to see if the target volume has been achieved
    const isEquilibriumAchieved = updateVolumeEquilibrium()

    if (!isEquilibriumAchieved) {
      // If equilibrium has not been achieved, set Mux Player to match volume value
      muxPlayerRef.current.volume = targetVolumeRef.current

      // Recursively call this function until equilibrium is achieved using a timeout
      // Recursion may not be needed but is here out of paranoia (with experience from YouTube player)
      timeoutId = setTimeout(enforcer, 100)
    }
  }

  // TODO: Find reliable way to know when Mux player is ready
  // @see https://github.com/muxinc/elements/issues/676
  setTimeout(enforcer, 500)

  return () => {
    if (timeoutId) {
      clearTimeout(timeoutId)
    }
    cancel = true
  }
}, [volume, src, updateVolumeEquilibrium])
```

Not a big fan of having to rely on timeouts and recursion if not necessary. That's why having a callback when the Mux Player UI is ready would be fantastic!

**Pull request merged**